### DISCUSSION TITLE
pest and pest++ output getters

### DISCUSSION THREAD

**Discussion by jtwhite79** [pypest/pyemu#557] [Ideas]
Been thinking about a helper class (or classes) that could be attached to `Pst` that would accept a master dir and then could lazy eval any number of known pest/pest++ output files.  Like for ies, you could do something like `pst.ies.prior_par_ensemble`, `pst.ies.iteration_3_obs_ensemble`, `pst.ies.actual_phi`, etc. Or for mou, `pst.mou.pareto_archive_summary` or `pst.mou.generation_10_dvpop`...The idea is that these items would only be loaded once and only loaded if requested.  This idea is only half baked but I keep seeing the same code being repeated across projects and this might be a way to reduce that need.

**Comment by laat0003** [comment]
Is this to simplify producing plots and tables for analysis and/or reporting?

**Reply by jtwhite79** [reply]
Really Just to simply accessing all the myriad of outputs produced.  I see these steps in every proj im inbolved with: load pst, lpad phi df,  load par and obs ensembles from either binary or ascii for several iterations, etc.  Basically just trying to minimize those repetitive bits of workflow code while also being smarter about what's held in memory...

**Reply by jtwhite79** [reply]
But could also lead to better/easier plotting
